source/fab/language/__init__.py

```diff
@@ -33,8 +33,8 @@ def products(self) -> List[Path]:
         raise NotImplementedError('Abstract methods must be implemented')
 
 
-class Analyser(Task):
-    def __init__(self, reader: TextReader, database: SqliteStateDatabase):
+class Analyser(Task, ABC):
```

Comment:
Interesting... I thought because `Task` is itself an `ABC` then the behaviour was somehow implicit?  So is this for clarity or does it actually change the behaviour?  (I'm happy with it, just curious)

Reply:
I'm not sure what the word of God is. Mypy and flake8 are both happy for abstractness to be inherited but PyCharm complains that `Analyser` does not implement all abstract methods, which of course it doesn't.

I think for clarity it is good to be explicit that this class is also abstract although I am open to the argument that the multiple inheritance approach taken here is inferior to an alternative.

Reply:
I think if the behaviour isn't affected by it then it doesn't hurt at all to be explicit.  And especially with PyCharm being quite a popular editor we are smoothing things for any other developers who come to work on Fab and use it.  I was really just curious if there was a practical reason for it that I wasn't aware of